File: typescript/src/PlanetTS1/Model/Player/Units/PSoldier.ts
import { ALocation as ALocation } from "../../Utilities"

export class PSoldier extends Soldier {
    loc : ALocation
    use : boolean
    drive : boolean
    owner : number
    icon : string

    constructor(location : ALocation) {
        super()
        this.loc = location
        this.use = true
        this.drive = true
        this.owner = 1
        this.icon = "pS"
    }
}

File: dotnet/src/Model/Player/Units/PSoldier.cs
using System;

namespace PlanetCS1 {
    class PSoldier : Soldier {
        public PSoldier(Location location) {
            this.loc = location;
            this.use = true;
            this.drive = true;
            this.owner = 1;
            this.icon = "pS";
        }
    }
}